src/features/hostapp/hooks/target-hostapp.ts

```diff
@@ -55,7 +58,46 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 			);
 		}
 	},
-});
+};
+
+const postHostappHook: hooks.Hooks = {
+	async POSTPARSE(args) {
+		if (
+			args.request.values.os_version != null &&
+			args.request.values.os_variant != null
+		) {
+			let deviceType;
+			if (!args.request.values.is_of__device_type) {
+				[deviceType] = await args.api.get({
+					resource: 'device_type',
+					options: {
+						$select: 'id',
+						$filter: {
+							slug: { $eq: args.request.values.device_type },
+						},
+					},
+				});
+			}
+			if (!args.request.values.is_of__device_type && !deviceType?.id) {
+				return;
+			}
+
+			const hostappRelease = await getOSReleaseResource(
+				args.api,
+				args.request.values.os_version,
+				args.request.values.os_variant,
+				args.request.values.is_of__device_type ?? deviceType?.id,
+			);
+			if (hostappRelease && hostappRelease.length > 0) {
+				args.request.values.should_be_operated_by__release =
+					hostappRelease[0].id;
```

Comment:
Can the `PATCH` hook do this the same way?

Reply:
the `PATCH` hook also enforces some extra constraints (not invalidated and not a downgrade), whereas the `POST` is less picky. do you mean specifically how the return value of `getOSReleaseResource` is used and validated?

Reply:
Yeah, the other bits can be specific to the PATCH as need be but this bit of setting the default seems like it should be the same for both?